pkg/controller/job/pod_failure_policy_test.go

```diff
@@ -702,6 +720,17 @@ func TestMatchPodFailurePolicy(t *testing.T) {
 			if tc.wantCountFailed != countFailed {
 				t.Errorf("Unexpected count failed. want: %v. got: %v", tc.wantCountFailed, countFailed)
 			}
+			if tc.wantAction == nil {
+				if action != nil {
+					t.Errorf("Unexpected job failure polic action. Got: %q", *action)
+				}
+			} else {
+				if action == nil {
+					t.Errorf("Missing job failure policy action. want: %q", *tc.wantAction)
+				} else if *tc.wantAction != *action {
+					t.Errorf("Unexpected job failure policy action. want: %v. got: %v", tc.wantAction, action)
+				}
+			}
```

Comment:
you can simplify all of this with `cmp.Equal` or `cmp.Diff`

Reply:
Indeed, used cmp.Diff for this and the check for the failure message